src/main/java/com/google/sps/servlets/DeleteRecipeServlet.java

```diff
@@ -16,11 +25,36 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long id = Long.parseLong(request.getParameter("id"));
+    long privateRecipeID = Long.parseLong(request.getParameter("id"));
+    String name = request.getParameter("name");
+    String description = request.getParameter("description");
 
-    // PrivateRecipes are the user uploaded recipes so only PrivateRecipes can be deleted (right now)
-    Key recipeEntitykey = KeyFactory.createKey("PrivateRecipe", id);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.delete(recipeEntitykey);
+
+    /// delete the private recipe
+    Key privateKey = KeyFactory.createKey("PrivateRecipe", privateRecipeID);
+    datastore.delete(privateKey);
+
+    // make filters to find the public recipe
+    Filter nameFilter = new FilterPredicate("name", FilterOperator.EQUAL, name);
```

Comment:
Why is this implementation preferred? Why not have the id of the public recipe saved as an attribute of the private recipe, this way you can just delete the recipe directly?

Reply:
yup that makes sense will do